packages/ra-ui-materialui/src/form/Toolbar.tsx

```diff
@@ -92,10 +78,8 @@ export const Toolbar = <
         >
             {Children.count(children) === 0 ? (
                 <div className={ToolbarClasses.defaultToolbar}>
-                    <SaveButton disabled={disabled} />
-                    {record && typeof record.id !== 'undefined' && (
```

Comment:
you need to keep this condition to hide the DeleteButton on create screens

Reply:
This condition already exists inside the `DeleteButton` itself, is duplicated here